src/integration/java/io/pinecone/integration/controlPlane/pod/CreateDescribeListAndDeleteIndexTest.java

```diff
@@ -1,51 +1,181 @@
+        assertEquals(createdIndex.getSpec().getPod().getPods(), 2);
+        assertEquals(createdIndex.getSpec().getPod().getPodType(), "p1.x2");
+        assertEquals(createdIndex.getStatus().getReady(), false);
+        assertEquals(createdIndex.getStatus().getState(), IndexModelStatus.StateEnum.INITIALIZING);
+
+        controlPlaneClient.deleteIndex(podIndexName);
+    }
+
+    @Test
+    public void createIndexWithInvalidName() {
+        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("Invalid-name")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("must consist of lower case alphanumeric characters or" +
+                    " '-'"));
+        }
+    }
+
+    @Test
+    public void createIndexWithInvalidDimension() {
+        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("invalid-dimension")
+                .metric(IndexMetric.COSINE)
+                .dimension(-1)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeUnmappedHttpException");
+        } catch (PineconeUnmappedHttpException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("dimension: invalid value"));
+        }
+    }
+
+    @Test
+    public void createIndexWithInvalidPods() {
+        CreateIndexRequestSpecPod podSpec =
+                new CreateIndexRequestSpecPod().environment(environment).pods(-1).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("invalid-pods")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("pods"));
```

Comment:
This might be a totally normal Java convention, so apologies if this is obvious, but why do you have an assert stmt for bot the `"pods"` part of the message and then the rest of it (`"must be greater than 0"`)?

Reply:
See my response above about the error assertions, I'm just checking for substrings in the larger error message.